### PR TITLE
fix: add burn funds command to console wallet (see issue #4547)

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Description
---
Adds command to burn funds on Tari console wallet.

Motivation and Context
---
Tackle issue #4547.

How Has This Been Tested?
---
Proper unit tests.

### CODE REVIEW

**Comment by jorgeantonio21** [comment]
Fixes #4547.

**Review by SWvheerden — COMMENTED** [review]
LGTM

File: applications/tari_console_wallet/src/automation/commands.rs [hunk 1]
```diff
@@ -609,6 +621,22 @@ pub async fn command_runner(
                     eprintln!("DiscoverPeer error! {}", e);
                 }
             },
+            BurnFunds(args) => {
+                match burn_tari(
```
Comment (hansieodendaal) [inline applications/tari_console_wallet/src/automation/commands.rs]: I think these two names should match - `BurnFunds` vs. `burn_tari`

**Review by hansieodendaal — COMMENTED** [review]
Looking good, but I believe we need a cucumber test for this:
- Let a miner connected to a wallet mine some blocks (20?)
- Mine some blocks (3?) without a connected wallet to have all funds available
- Verify the miner wallet's available balance (and possibly the balance of the entire UTXO set)
- Create and submit a burn funds transaction to burn (almost) all funds in the wallet
- Mine some blocks (3?) without a connected wallet
- Verify the balance of the entire UTXO set

File: applications/tari_console_wallet/src/automation/commands.rs (hunk 1 above)
Comment (jorgeantonio21) [reply applications/tari_console_wallet/src/automation/commands.rs]: Definitely ! Thanks for the suggestions

**Comment by stringhandler** [comment]
Happy to merge, but let's add the cucumber test first

**Comment by SWvheerden** [comment]
We have cucumber tests for the burns, verify balance etc.
We just need one for the cli

Look at: 
`WalletTransactions.feature - Create burn transaction`
`Sync.feature - Sync burned output`
`Sync.feature - Pruned node should handle burned output`

File: integration_tests/features/WalletCli.feature [hunk 2]
```diff
@@ -53,6 +53,21 @@ Feature: Wallet CLI
         And mining node MINE mines 5 blocks
         Then I wait for wallet RECEIVER to have at least 1000000 uT
 
+    @long-running
```
Comment (stringhandler) [inline integration_tests/features/WalletCli.feature]:
```suggestion
  
```

**Pull request merged**